homeassistant/components/netgear/config_flow.py

```diff
@@ -0,0 +1,190 @@
+        """Init object."""
+        self.config_entry = config_entry
+
+    async def async_step_init(self, user_input=None):
+        """Manage the options."""
+        if user_input is not None:
+            return self.async_create_entry(title="", data=user_input)
+
+        settings_schema = vol.Schema(
+            {
+                vol.Optional(
+                    CONF_CONSIDER_HOME,
+                    default=self.config_entry.options.get(
+                        CONF_CONSIDER_HOME, DEFAULT_CONSIDER_HOME.total_seconds()
+                    ),
+                ): int,
+                vol.Optional(
+                    CONF_TRACKED_LIST,
+                    description={
+                        "suggested_value": self.config_entry.options.get(
+                            CONF_TRACKED_LIST, ""
+                        )
+                    },
+                ): str,
+            }
+        )
+
+        return self.async_show_form(step_id="init", data_schema=settings_schema)
+
+
+class NetgearFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow."""
+
+    VERSION = 1
+
+    def __init__(self):
+        """Initialize the netgear config flow."""
+        self.placeholders = {
+            CONF_NAME: "",
+            CONF_HOST: DEFAULT_HOST,
+            CONF_PORT: DEFAULT_PORT,
+            CONF_USERNAME: DEFAULT_USER,
+        }
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> OptionsFlowHandler:
+        """Get the options flow."""
+        return OptionsFlowHandler(config_entry)
+
+    async def _show_setup_form(self, user_input=None, errors=None):
+        """Show the setup form to the user."""
+        if not user_input:
+            user_input = {}
+
+        if self.placeholders.get(CONF_URL):
+            user_input.update({CONF_URL: self.placeholders[CONF_URL]})
+            data_schema = _discovery_schema_with_defaults(user_input)
+        else:
+            data_schema = _user_schema_with_defaults(user_input)
+
+        return self.async_show_form(
+            step_id="user",
+            data_schema=data_schema,
+            errors=errors or {},
+            description_placeholders=self.placeholders,
+        )
+
+    async def async_step_import(self, user_input=None):
+        """Import a config entry."""
+        return await self.async_step_user(user_input)
+
+    async def async_step_ssdp(self, discovery_info: dict):
+        """Initialize flow from ssdp."""
+        await self.async_set_unique_id(discovery_info[ssdp.ATTR_UPNP_SERIAL])
+        self._abort_if_unique_id_configured()
+
+        self.placeholders[CONF_NAME] = discovery_info[ssdp.ATTR_UPNP_MODEL_NUMBER]
+        self.placeholders[
+            CONF_URL
+        ] = f"http://{urlparse(discovery_info[ssdp.ATTR_SSDP_LOCATION]).hostname}/"
+        return await self._show_setup_form()
```

Comment:
We should only display the form for the step in the step.

Move to the user step by calling `async_step_user` or create a new step.

Reply:
done